src/Context/TodosContext.tsx

```diff
@@ -0,0 +1,71 @@
+import { createContext } from 'react';
+import { TodoContext } from '../types/TodoContext';
+import { useState } from 'react';
+import { Errors } from '../types/Errors';
+import { FilterTodos } from '../types/FilterTodos';
+import { Todo } from '../types/Todo';
+import * as todoSevice from '../api/todos';
+import { handleRequestError } from '../utils/handleRequestError';
+
+export const TodosContext = createContext<TodoContext | undefined>(undefined);
+
+type Props = {
+  children: React.ReactNode;
+};
+
+export const TodosContextProvider: React.FC<Props> = ({ children }) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterSelected, setFilterSelected] = useState<FilterTodos>(
+    FilterTodos.all,
+  );
+  const [error, setError] = useState<Errors>(Errors.default);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsloading] = useState(true);
+
+  const completedTodos = todos.filter(todo => todo.completed);
+  const activeTodos = todos.filter(todo => !todo.completed);
+
+  function onDelete(todoId: number) {
+    setLoadingTodoIds([todoId]);
+
+    todoSevice
+      .deleteTodo(todoId)
+      .then(() => {
+        setTodos(currentTodos =>
+          currentTodos.filter(todo => todo.id !== todoId),
+        );
+        setIsloading(true);
+      })
+      .catch(() => {
+        handleRequestError(Errors.deleteTodo, setError);
+        setLoadingTodoIds([]);
+        setIsloading(true);
+      });
+    setLoadingTodoIds([todoId]);
```

Comment:
```suggestion
      .then(() => {
        setTodos(currentTodos =>
          currentTodos.filter(todo => todo.id !== todoId),
        );
      })
      .catch(() => {
        handleRequestError(Errors.deleteTodo, setError);
      })
      .finally(() => {
        setLoadingTodoIds(prevLoadingTodoIds => (
          prevLoadingTodoIds.filter(id => id !== todoId)
        ));
      });
```
- You have a state with loading todo ids, why do you need the second state `isLoading`?
- You should remove the loading state in the finally block as you do the same in both cases.
- You already set the loader before fetching, you don't need it to do twice